orca-clouddriver/src/main/java/com/netflix/spinnaker/orca/clouddriver/pipeline/manifest/DeployManifestStage.java

```diff
@@ -90,81 +104,214 @@ public void afterStages(@Nonnull StageExecution stage, @Nonnull StageGraphBuilde
-      String clusterName,
-      String namespace,
-      String newManifestName) {
-    return oortService
-        .getClusterManifests(account, namespace, "replicaSet", application, clusterName)
-        .stream()
-        .filter(m -> !m.getFullResourceName().equals(newManifestName))
-        .map(ManifestCoordinates::getFullResourceName)
-        .collect(toImmutableList());
+  /**
+   * Delegate class to handle all manifest-related actions in this file such as fetching manifest
+   * from external service or extracting manifest params from parentContext
+   */
+  @Component
+  @RequiredArgsConstructor
+  static class ManifestOperationsHelper {
+
+    private static final String REPLICA_SET = "replicaSet";
+    private static final String KIND = "kind";
+    private static final String OUTPUTS_MANIFEST = "outputs.manifests";
+
+    private final OortService oortService;
+
+    /**
+     * Returns all replicaSet manifests from the cluster. Search is performed in external service,
+     * search parameters match manifest deployed in current stage.
+     *
+     * @param dm - deployment manifest of current stage
+     * @return list of all manifest already deployed to the cluster
+     */
+    ImmutableList<String> getOldManifestNames(DeployedManifest dm) {
+      return oortService
+          .getClusterManifests(
+              dm.getAccount(),
+              dm.getNamespace(),
+              REPLICA_SET,
+              dm.getApplication(),
+              dm.getClusterName())
+          .stream()
+          .filter(m -> !m.getFullResourceName().equals(dm.getManifestName()))
+          .map(ManifestCoordinates::getFullResourceName)
+          .collect(toImmutableList());
+    }
+
+    /**
+     * Returns replicaSet manifests from the {@code parentContext}
+     *
+     * @param parentContext of currently processed stage
+     * @return list of replicaSet manifests deployed in the cluster - obtained directly from the
+     *     {@code parentContext}
+     */
+    List<Map<String, ?>> getNewManifests(Map<String, Object> parentContext) {
+      var manifestsFromParentContext = (List<Map<String, ?>>) parentContext.get(OUTPUTS_MANIFEST);
+      return manifestsFromParentContext.stream()
+          .filter(manifest -> REPLICA_SET.equalsIgnoreCase((String) manifest.get(KIND)))
+          .collect(Collectors.toList());
+    }
+
+    /**
+     * During a B/G deployment, if the blue deployment fails to create pods (due to issues such as
+     * an incorrect image name), the deployment will not fail, but will wait indefinitely to achieve
+     * stability. This is indicated by status.failed=false and status.stable=false. This method
+     * checks for such a situation.
+     *
+     * @param account used to run deployment
+     * @param name of the manifest
+     * @return true, if manifest was not deployed correctly and waits to get stable, false otherwise
+     */
+    boolean previousDeploymentNeitherStableNorFailed(String account, String name) {
```

Comment:
This method is the actual fix to the bug. We need to very previous manifest status.